### ISSUE TITLE
Cross site image link does not work (also for merger sites)

### ISSUE THREAD

**Issue by slrslr** [HelloZeroNet/ZeroNet#1992]
Rev3860, when i add to my ZeroBlog an image linked from external site:

![text](/otherziteaddress/data/users/userid/image.jpg)

then image not show up until (i think) i visit that site at least once ?
I hope you can make the zeronet download linked file/site automaticaly when it is linked like that.

It is similar with Millchan hubs. When i not visit the hub the link to the merger site, which points to a content on that hub will show Not Found error. I think this is important thing to the TODO list to fix.

**Comment by HelloZeroNet** [comment]
This is intentional: Otherwise if someone create a page that has 100 files from 100 different sites, then you would end up serving 100 more sites.

**Comment by slrslr** [comment]
@HelloZeroNet  I do not see anything bad on ZN to download 100s of sites, if zeronet is intended to be used by active users, it should be able to host 1000s of sites on regular or low end computer.

If 100s of sites is a problem, then maybe create a limit on amount of requests one page can create or ask confirmation from user to download external site. Showing fake message "Not found" is not good, because content exist, only zeronet is not made to show it. I think cross content linking is good, why to build duplicity.

**Comment by HelloZeroNet** [comment]
Unfortunately it's not possible to find out which browser window did the request, so it's not possible to show confirmation. To be able to receive site updates you need at least 5 connection per site, which can be problematic with 1000s of sites. 
The solution to this problem could be adding non-mutable file support. For that files the client would not need to keep open connections, because the would not be updates for the files.

**Comment by krixano** [comment]
I think we should **not** let zites *automatically* download other zites - especially since every zite you download is one you seed. It should show a confirmation thing at the very least (which zites can already do I believe by using a ZeroFrame command, not sure though).

**Comment by slrslr** [comment]
@Krixano you are right, though interlinking images and videos is imporant. Uploading duplicate/already posted images/videos seems waste/stupid in many cases.

Unsure if there can be made the way *zeronet download only linked media content / json files, instead of full site*. It may be made even the way such downloaded site will be downloaded only temporarily and maybe not seeded.

**Comment by imachug** [comment]
This might actually become a problem - if a malicious site links to hundreds of different zites, all with 10MB images, you have to download several gigabytes.

**Comment by slrslr** [comment]
@imachug 
> hundreds of different zites

possibly zeronet can be made to connect maximum let's say 20 external sites per page load.

>  you have to download several gigabytes.

I do not think so, because 
1. number of external hot-linked images (jpeg|jpg|png|gif|webp) loaded from single page can(?) be limitted to something like 70
2. page visit may not take enough time to download that much data and the third party site may(?) be made to stop downloading (& possibly delete with some timeout) once parent site (that linked to a image) was closed
3. zeronet would download external site special way - only download things like maybe content.json and the requested image/video file (known extensions). 
4. unsure if zeronet can show one prompt per external site triggered and say: "Can this site download 20 image files (123 MB) from external site? OK" else user would have to click the image to load manually (like on ZeroMe).

**Comment by slrslr** [comment]
"ZeroBlog++" has the feature to ask visitor if he want to grant the permission to download external site images, but unfortunately if the external site is bigger than 10MB, i think it may break/stop loading of image/s. I increased the site size a few times clicking button on ZeroHello and refreshed blog page to see external images. That is bad. Zeronet should load images hotlinked from external sites better way.
